### PR TITLE
KAA-570: [Java SDK] Add warning messages as a result of API calls when client is in invalid state.

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

File: client/client-multi/client-java-core/src/main/java/org/kaaproject/kaa/client/AbstractKaaClient.java [hunk 1]
```diff
@@ -205,12 +230,15 @@
...
         transportContext.initTransports(this.channelManager, this.kaaClientState);
-        
+
         eventFamilyFactory = new EventFamilyFactory(eventManager, context.getExecutorContext());
     }
 
     @Override
-    public void start() {
+    public void start() throws KaaException {
+        checkClientStateNot(State.STARTED, "Kaa client is already run");
```
Comment (ashvayka) [inline client/client-multi/client-java-core/src/main/java/org/kaaproject/kaa/client/AbstractKaaClient.java]: is already started?

**Comment by batytskyy** [comment]
Do we really need to change Kaa Client SDK API declaring that KaaClient's methods throw KaaException?

**Comment by ashvayka** [comment]
LGTM

File: client/client-multi/client-java-core/src/main/java/org/kaaproject/kaa/client/AbstractKaaClient.java [hunk 2]
```diff
@@ -155,6 +155,31 @@
...
     protected final KaaClientStateListener stateListener;
 
+    protected enum State {
+        CREATED,
+        STARTED,
+        PAUSED,
+        STOPPED
+    };
+
+    protected State clientState = State.CREATED;
```
Comment (shpigor) [inline client/client-multi/client-java-core/src/main/java/org/kaaproject/kaa/client/AbstractKaaClient.java:167]: Maybe better to move to the end of constructor

File: client/client-multi/client-java-core/src/main/java/org/kaaproject/kaa/client/AbstractKaaClient.java [hunk 3]
```diff
@@ -375,46 +420,51 @@ public void removeNotificationListener(String topicId, NotificationListener list
 
     @Override
     public void subscribeToTopic(String topicId) throws UnavailableTopicException {
-        this.notificationManager.subscribeToTopic(topicId, true);
+        subscribeToTopic(topicId, true);
```
Comment (shpigor) [inline client/client-multi/client-java-core/src/main/java/org/kaaproject/kaa/client/AbstractKaaClient.java]: I think better to use constants FORCE_SYNC and WITHOUT_SYNC instead bool value.

**Comment by shpigor** [comment]
Revieved

**Pull request merged**